lab_solutions/lab03_3_rational_number_class.py

```diff
@@ -0,0 +1,166 @@
+#! /usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright © 2020 Attila Nagy <attila.nagy234@gmail.com>
+#
+# Distributed under terms of the MIT license.
+
+
+def gcd(x, y):
+    while x % y != 0:
+        prev_x = x
+        prev_y = y
+        x = prev_y
+        y = prev_x % prev_y
+    return y
+
+
+class RationalNumberValueError(ValueError):
+    pass
+
+
+class RationalNumber:
+    def __init__(self, p, q=1):
+        self.validate_numerator(p)
+        self.validate_denominator(q)
+        greatest_common_div = gcd(p, q)
+        if greatest_common_div != 1:
+            self._p = p // greatest_common_div
+            self._q = q // greatest_common_div
+        else:
+            self._p = p
+            self._q = q
+        if self._q < 0:
+            self.switch_signs()
+
+    def switch_signs(self):
+        self._p *= -1
+        self._q *= -1
+
+    def cross_multiple(self, other):
+        return self.p * other.q + self.q * other.p
+
+    def __add__(self, other):
+        numerator = self.cross_multiple(other)
+        denominator = self.q * other.q
+        return RationalNumber(numerator, denominator)
+
+    def __eq__(self, other):
+        return (self.p * other.q) == (self.q * other.p)
```

Comment:
```suggestion
        return (self.p == other.p) and (self.q == other.p)
```

This works only if you fix the setters.